### PR TITLE
Admin validation

### PR DESCRIPTION
**Pull request by yuvalelarat**
Added validateAdmin middleware on "/api/validate-admin"

### CODE REVIEW

**Comment by yuvalelarat** [comment]
Please notice me if its good enough

File: src/index.ts [hunk 1]
```diff
@@ -16,6 +17,7 @@ app.use(express.json());
 app.use("/login", loginRouter);
 app.use("/register", registerRouter);
 app.use("/api", authenticateUser);
+app.use("/api/validate-admin", authenticateUser, validateAdmin);
```
Comment (tomered) [inline src/index.ts:20]: line 19 should look like this:
app.use("/api", authenticateUser, authenticateUser, validateAdmin, adminEndpoint);
delete line 20 and create adminEndpoint

File: src/middleware/validateAdmin.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Request, Response, NextFunction } from "express";
+import { verifyJwt } from "../utils/jwt";
+import { JwtPayload } from "jsonwebtoken";
+import { User } from "../entities/user";
+
```
Comment (tomered) [inline src/middleware/validateAdmin.ts:5]: please add documentation and comments

File: src/middleware/validateAdmin.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      data: {
+        userName: user.userName,
+        role: user.role.roleId,
+      },
+    });
+  } catch (err) {
+    console.error(err);
+    return res.status(500).send("Internal server error.");
+  }
+};
```
Comment (tomered) [inline src/middleware/validateAdmin.ts:65]: add next to middleware function

File: src/index.ts (hunk 1 above)
Comment (tomered) [inline src/index.ts]: line 19 should look like this:
app.use("/api", authenticateUser, authenticateUser, validateAdmin);

**Review by tomered — CHANGES_REQUESTED** [review]

**Comment by yuvalelarat** [comment]
I have added adminEndpoint.ts in routes folder and changed to app.use("/api", authenticateUser, validateAdmin, adminEndpoint); + added comments
is that what you meant?

**Comment by yuvalelarat** [comment]
done.

**Review by tomered — APPROVED** [review]

**Pull request merged**